validate-xml/validate-xml.cpp

```diff
@@ -0,0 +1,372 @@
+
+        entityFileResolver.AddFileContent("SMPTE_CPL_SCHEMA_NAME", CPL_429_7_2006_XSD,
+                                          CPL_429_7_2006_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("XML_DSIG_SCHEMA_NAME", XMLDSIG_CORE_SCHEMA_XSD,
+                                          XMLDSIG_CORE_SCHEMA_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("SMPTE_CPL_STEREO_SCHEMA_NAME",
+                                          MAIN_STEREO_PICTURE_CPL_429_10_2008_XSD,
+                                          MAIN_STEREO_PICTURE_CPL_429_10_2008_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("INTEROP_CPL_STEREO_SCHEMA_NAME",
+                                          MAIN_STEREO_PICTURE_CPL_XSD,
+                                          MAIN_STEREO_PICTURE_CPL_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("INTEROP_CPL_CC_SCHEMA_NAME",
+                                          PROTO_ASDCP_CC_CPL_20070926_XSD,
+                                          PROTO_ASDCP_CC_CPL_20070926_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("SMPTE_TIMED_TEXT_SCHEMA_NAME", TT_429_12_2008_XSD,
+                                          TT_429_12_2008_XSD_SIZE);
+
+        entityFileResolver.AddFileContent("CPL_DOLBY_ATMOS_AUX_DATA_SCHEMA_NAME",
+                                          DOLBY_AD_2012_XSD,
+                                          DOLBY_AD_2012_XSD_SIZE);
+    }
+
+    parser->setEntityResolver(&entityFileResolver);
+
+    try
+    {
+        parser->parse(argv[1]);
+    }
+    catch (const OutOfMemoryException&)
+    {
+        cerr << "Out of memory exception." << endl;
+		return schemaError;
+    }
+    catch (const XMLException& e)
+    {
+        cerr << "An error occurred during parsing" << endl
+             << "   Message: " << StrX(e.getMessage()) << endl;
+		return schemaError;
+    }
+    catch (const DOMException& e)
+    {
+        const unsigned int maxChars = 2047;
+        XMLCh errText[maxChars + 1];
+        cerr << endl
+             << "A DOM error occurred during parsing: '"
+             << std::string(argv[1])
+             << "'" << endl
+             << "DOM Exception code:  "
+             << e.code << endl;
+        if (DOMImplementation::loadDOMExceptionMsg(e.code, errText, maxChars))
+        {
+            cerr << "Message is: " << StrX(errText) << endl;
+        }
+		return schemaError;
+    }
+    catch (...)
+    {
+        cerr << "An unclassified error occurred during parsing." << endl;
+		return schemaError;
+    }
+
+	DOMNode *doc;	
+	doc = parser->getDocument();
+	DOMDocument *theDOM = parser->getDocument();
+	// Find the signature node
+	DOMNode *sigNode = findDSIGNode(doc, "Signature");
+
+	if (sigNode == 0)
+	{
+		DOMElement *FirstNode = theDOM->getDocumentElement();
+		const char* root = XMLString::transcode(FirstNode->getNodeName());
+		if(strcmp(root,"DCinemaSecurityMessage"))
+		{
+			cout<<"Valid XML"<<endl;
```

Comment:
@sivachandran strcmp returns 0 when strings are equal, so the else condition will only be executed in that case of KDM. Instead of making three compares for CPL, PKL and AssetMap I have done like this.